acstools/acszpt.py

```diff
@@ -0,0 +1,329 @@
+        results.
+
+        """
+        soup = BeautifulSoup(
+            self._response.content.decode(self._response.encoding),
+            'html.parser'
+        )
+        # Grab all elements in the table returned by the ZPT calc.
+        td = soup.find_all('td')
+        for j in range(6):
+            for i, val in enumerate(td[j::6]):
+                if not i:
+                    self._zpts[val.text.split('[')[0].strip()] = []
+                else:
+                    self._zpts[list(self._zpts.keys())[j]].append(val.text)
+
+    def _format_results(self):
+        """ Format the results into an astropy.table.Table with corresponding
+        units
+
+        Returns
+        -------
+        astropy.table.Table object with the zeropoints for the supplied date
+        and detector
+        """
+        data_types = []
+        data_units = []
+        for i, key in enumerate(self._zpts.keys()):
+            if key.lower() == 'filter':
+                data_types.append(str)
+                data_units.append(u.dimensionless_unscaled)
+
+            elif key.lower() == 'photplam':
+                data_types.append(float)
+                data_units.append(u.angstrom)
+
+            elif key.lower() == 'photflam':
+                data_types.append(float)
+                data_units.append(u.erg / u.cm**2 / u.second / u.angstrom)
+
+            elif key.lower() == 'stmag':
+                data_types.append(float)
+                data_units.append(u.STmag)
+
+            elif key.lower() == 'vegamag':
+                data_types.append(float)
+                data_units.append(u.mag)
+
+            elif key.lower() == 'abmag':
+                data_types.append(float)
+                data_units.append(u.ABmag)
+
+        tab = Table(list(self._zpts.values()),
+                    names=list(self._zpts.keys()),
+                    dtype=data_types)
+
+        # Loop through each column and set the appropriate units
+        for i, col in enumerate(tab.colnames):
+            tab[col].unit = data_units[i]
+        self.zpt_table = tab
+
+    def fetch(self):
+        """ This method is used to submit a request to the ACS Zeropoints Calculator
+
+        Submit the request and format the results into an Astropy Table object
+        with the correspond units applied to each column
+
+        Returns
+        -------
+        astropy.table.Table or None
+            Returns a table if the request was successful and None otherwise
+        """
+        valid_inputs = self._check_inputs()
+        if valid_inputs:
+            self._submit_request()
+            if self._failed:
+                return None
+            else:
+                self._parse_response()
+                self._format_results()
+                return self.zpt_table
+        else:
+            print('Please fix the incorrect input(s)')
+            return None
+
+
+if __name__ == '__main__':
```

Comment:
This looks like leftover code for debugging. Please remove. Or you can turn this into a proper test but have Jenkins skip it by default, or use mock to avoid network calls. The latter is preferred but more work.

Reply:
If you're referring to the `if __name__ == '__main__':`, then yeah it is leftover for debugging and I'll remove it